sulis_module/src/loot_list.rs

```diff
@@ -301,47 +285,37 @@ impl LootList {
 
     fn gen_adjectives(&self, entry: &Entry) -> Vec<String> {
         let mut result = Vec::new();
-        if entry.adjective1_total_weight > 0 {
-            let roll = gen_rand(0, entry.adjective1_total_weight);
-
-            let mut cur_weight = 0;
-            for (id, weight) in entry.adjective1.iter() {
-                cur_weight += weight;
-                if roll < cur_weight {
-                    result.push(id.clone());
-                    break;
-                }
-            }
-        }
 
-        if entry.adjective2_total_weight > 0 {
-            let roll = gen_rand(0, entry.adjective2_total_weight);
+        if !entry.adjective1.is_empty() {
+            let pick = gen_rand_weight(&entry.adjective1);
+            result.push(pick.clone());
+        }
 
-            let mut cur_weight = 0;
-            for (id, weight) in entry.adjective2.iter() {
-                cur_weight += weight;
-                if roll < cur_weight {
-                    result.push(id.clone());
-                    break;
+        if !entry.adjective2.is_empty() {
+            let mut remain = 1;
+            while remain > 0 {
+                let pick = gen_rand_weight(&entry.adjective2);
+                match pick.as_str() {
+                    "unidentified" => {
```

Comment:
i do not like how "unidentified" is hard coded here to only appear in adjective2.  it shouldn't need special treatment here, if it were to just appear in its own regular, mutually exclusive category.  such as adjective3.

Really, adjectives should be an array of arbitrary length.  But that doesn't need to be fixed in this PR.

Reply:
I have it fixed anyway. adjective[123] -> adjectives.